### ISSUE TITLE
Racks Plugin - Can't assign computers

### ISSUE THREAD

**Issue by hgpit** [InfotelGLPI/racks#38]
Hi,
We're using GLPI v0.90.1 and have just installed the Rack Enclosures plugin (latest build from GitHub).

It looks really good but I am unable to assign any computers to a rack. I noticed a similar problem here http://forum.glpi-project.org/viewtopic.php?id=17908 but my rack and computers are in the same entity. Regardless of this, I have tried allowing child entities in my rack, but it's the same problem.

Basically, when I choose the 'Front' tab in my rack asset management, then choose 'computers' from the 'Add equipment' drop-down, there are no computers to choose from; neither can I search for any.

Please can anyone point out what I might be doing wrong or overlooking?

Thank you!

Best Regards,

Elliot

**Comment by tsmr** [comment]
Can you test solution of #22

**Comment by hgpit** [comment]
Hi, thanks for the reply!

I have followed the solution you referenced, but don't appear to have any joy.

On line 138-139 of my ajax/dropdownValue.php file I have:

`'text'  => Toolbox::substr($data["name"], 0, 
                                                          50));`

I have replaced these lines with:

`'text'  => substr($data["name"], 0, 
                                                     $CFG_GLPI["dropdown_chars_limit"]));`

as suggested, but I appear to get the same results.

This is a shame, as it was sounding really promising!

**Comment by hgpit** [comment]
Here's a screenie of the issue on our GLPI (I have noticed it presents differently to some of the previous posters who have raised related issues).
![racks_plugin_error](https://cloud.githubusercontent.com/assets/16598973/18592293/3f9be5fc-7c2e-11e6-99cc-fa8bd950e547.jpg)

**Comment by hgpit** [comment]
Sorry, I just noticed the part which says there's no "$CFG_GLPI". But even with the following, I still have no populated drop-down:

`'text'  => Toolbox::substr($data["name"], 0,
                                                     $_SESSION["glpidropdown_chars_limit"]));`

**Comment by hgpit** [comment]
It appears I was being a dufus!
I just noticed in one of the other posts, a link to http://glpi/plugins/racks/front/itemspecification.php. It wasn't obvious to me that it was a pre-requisite to configure 'specifications' for different asset models - although it does seem more obvious now!

I have configured some specs and re-instated the original ajax/dropdownValue.php code and I now have some assets in my dropdown lists :)

Thank you!
